File: m/rls/strev.r
file	strev - reverse string
include	rid:rider
include	rid:stdef

code	st_rev - reverse string

;	a bcde f	a bcd e
;	f bcde a	e bcd a
;	fe cd ba	ed c ba

  proc	st_rev
	str : * char
	()  : * char			; past string
  is	res : * char			; result
	lst : * char			; last 
	cha : int			; temp
	reply str if *str eq		; null string
	lst = res = st_end (str)	; point at end
	while str lt --lst		; got more
	   cha = *lst			; get temp
	   *lst = *str			; swap another
	   *str++ = cha			;
	end				;
	reply res			; end of it
  end


File: m/rls/rlb/strev.c
/* file -  strev - reverse string */
#include "m:\rid\rider.h"
#include "m:\rid\stdef.h"
/* code -  st_rev - reverse string */
char *st_rev(
char *str )
{ char *res ;
  char *lst ;
  int cha ;
  if( *str == 0)return ( str );
  lst = res = st_end (str);
  while ( str < --lst) {
    cha = *lst;
    *lst = *str;
    *str++ = cha;
  } 
  return ( res);
} 
